scipy/signal/tests/test_fir_filter_design.py

```diff
@@ -392,6 +392,18 @@ def test_fs_nyq(self):
         taps2 = firwin2(80, [0.0, 30.0, 60.0], [1.0, 1.0, 0.0], nyq=60.0)
         assert_array_almost_equal(taps1, taps2)
 
+    def test_tuple(self):
+        taps1 = firwin2(150, (0.0, 0.5, 0.5, 1.0), (1.0, 1.0, 0.0, 0.0))
+        taps2 = firwin2(150, [0.0, 0.5, 0.5, 1.0], [1.0, 1.0, 0.0, 0.0])
+        assert_array_almost_equal(taps1, taps2)
+
+    def test_input_modyfication(self):
+        freq1 = np.array([0.0, 0.5, 0.5, 1.0])
+        freq2 = np.array(freq1)
+        firwin2(80, freq1, [1.0, 1.0, 0.0, 0.0])
+        assert_equal(freq1, freq2)
```

Comment:
You added some degenerate cases, so it would be good to add corresponding tests like:
```
with pytest.raises(ValueError, match='cannot contain numbers that are closer':
    firwin2(...)
```
To ensure we are catching all cases properly, there should probably be one of these for each `raise ...` lines in the function (including existing ones, which might or might not already be tested -- please check and add if necessary)

Reply:
I have now added missing tests in the form in which the existing ones were written. Therefore they only check whether the exception has been raised. If you want I could rewrite all assertions so that they also check if error message is the expected one.

Reply:
You don't have to update the old ones (though you can if you want). But new code should follow current best practice, even though it might lead to inconsistencies in test implementation/thoroughness in the file.